src/api-next/api-next.ts

```diff
@@ -1200,13 +1210,73 @@ export class IBApiNext {
         [EventName.tickSize, this.onTick],
         [EventName.tickGeneric, this.onTick],
         [EventName.tickOptionComputation, this.onTickOptionComputation],
+        [EventName.tickSnapshotEnd, this.onTickSnapshotEnd],
       ],
       snapshot || regulatorySnapshot
         ? undefined
         : `${JSON.stringify(contract)}:${genericTickList}`
     );
   }
 
+  /**
+   * Fetch a snapshot of real time market data.
+   * Returns market data for an instrument either in real time or 10-15 minutes delayed (depending on the market data type specified,
+   * see [[setMarketDataType]]).
+   * getMarketDataSingle will collect market data for a maximum of 11 seconds and then return the result.
+   *
+   * @param contract The [[Contract]] for which the data is being requested
+   * @param genericTickList comma  separated ids of the generic ticks
+   * Look at getMarketData documentation for a list of available generic ticks.
+   * @param snapshot For users with corresponding real time market data subscriptions.
+   * A `true` value will return a one-time snapshot, while a `false` value will provide streaming data.
+   * @param regulatorySnapshot Snapshot for US stocks requests NBBO snapshots for users which have "US Securities Snapshot Bundle" subscription
+   * but not corresponding Network A, B, or C subscription necessary for streaming * market data.
+   * One-time snapshot of current market price that will incur a fee of 1 cent to the account per snapshot.
+   */
+  getMarketDataSingle(
```

Comment:
Since getMarketData now handles tickSnapshotEnd by completing the Observable, this can shrink to:
```suggestion
  getMarketDataSingle(
    contract: Contract,
    genericTickList: string,
    regulatorySnapshot: boolean
  ): Promise<MarketDataTicks> {
    return lastValueFrom(
      this.getMarketData(
        contract,
        genericTickList,
        true,
        regulatorySnapshot
      ).pipe(map((v: { all: MutableMarketData }) => v.all)),
      {
        defaultValue: new MutableMarketData(),
      }
    );
  }
```